pkg/api/handlers/compat/images_build.go

```diff
@@ -134,6 +134,13 @@ func BuildImage(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	// if layers field not set assume its not from a valid podman-client
+	// could be a docker client, set `layers=true` since that is the default
+	// expected behviour
+	if _, found := r.URL.Query()["layers"]; !found {
```

Comment:
That's not enough.  We need to check whether the handler was called via libpod or compat. Otherwise we depend on behavior that may change.

Wrap that into `if !utils.IsLibpodRequest(r) { ..}`

Reply:
Why should layers not default to true in the libpod API? podman build defaults to true so should the API IMO.

Reply:
@Luap99 i guess because `podman-remote` has a option to configure it as `false` so we must honor it.

Reply:
@vrothberg ackd.

Reply:
We will always honor it, this is just the default.

Reply:
@vrothberg @mheon Why should the API have a different default then the podman cli?

Reply:
Tests broke so there is an expectation on these libpod defaults.  Changing them seems like a separate issue to me.